customer-models/src/main/java/org/powertac/customer/evcharger/DemandSampler.java

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (c) 2022 by John Collins.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.powertac.customer.evcharger;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import org.apache.commons.configuration2.XMLConfiguration;
+import org.apache.commons.configuration2.ex.ConfigurationException;
+import org.apache.commons.math3.distribution.MixtureMultivariateNormalDistribution;
+import org.apache.commons.math3.distribution.MultivariateNormalDistribution;
+import org.apache.commons.math3.distribution.NormalDistribution;
+import org.apache.commons.math3.linear.Array2DRowRealMatrix;
+import org.apache.commons.math3.util.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.powertac.common.config.Configurator;
+
+/**
+ * Sampler class which holds the logic of sampling new EV charger demand from
+ * the statistical model.
+ *
+ * @author Philipp Page <github@philipp-page.de>
+ */
+class DemandSampler
+{
+  private static final Logger log = LogManager.getLogger(DemandSampler.class.getSimpleName());
+  private MixtureMultivariateNormalDistribution pluginProbability;
+  private HashMap<String, MixtureMultivariateNormalDistribution> condHorizonDemandProbabilities = new HashMap<>();
+  private Integer seed;
+  private XMLConfiguration config;
+
+  void initialize ()
+  {
+    // TODO: Make this dynamic (How?)
+    String path = "config/residential_ev.xml";
+    try {
+      config = Configurator.readXML(path);
+    }
+    catch (ConfigurationException e) {
+      log.error("Problem loading configuration: " + e);
```

Comment:
We cannot recover if the configuration cannot be found. How shall we deal with this?

Reply:
For the time being, logging an error is almost enough. We also need to either (a) prevent the model from running, (b) use some default values, or (c) crash the server. If (b) is not practical, then (a) is OK. It's easy to prevent the model from running, just check a flag at the start of the step() and evaluateTariffs() methods.